openedx/core/djangoapps/content/block_structure/tests/test_models.py

```diff
@@ -135,15 +133,14 @@ def test_prune_keep_count(self, prune_keep_count, num_prior_edits):
             'openedx.core.djangoapps.content.block_structure.config.num_versions_to_keep',
             return_value=prune_keep_count,
         ):
-            for _ in range(num_prior_edits):
-                self._verify_update_or_create_call('data')
+            for x in range(num_prior_edits):
+                self._verify_update_or_create_call('data_{}'.format(x))
 
             if num_prior_edits:
-                self._assert_file_count_equal(num_prior_edits)
+                self._assert_file_count_equal(min(num_prior_edits, prune_keep_count))
```

Comment:
This method is slightly changed to reflect the fact that `PRUNING_ACTIVE` is a configuration variable that is not expected to change during execution. Previously, the fact that the first half of this test did *not* have the flag active and the serialized data was always `data`, which caused counting issues. I've updated to use varying serialized data and handle the expectation that the entire test is runs with pruning enabled.